api/session.go

```diff
@@ -2,16 +2,121 @@ package api
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 
 	"github.com/manifoldco/torus-cli/apitypes"
+	"github.com/manifoldco/torus-cli/base64"
+	"github.com/manifoldco/torus-cli/envelope"
+	"github.com/manifoldco/torus-cli/identity"
+	"github.com/manifoldco/torus-cli/primitive"
 )
 
+type session struct {
+	sessionType string
+	identity    *envelope.Unsigned
+	auth        *envelope.Unsigned
+}
+
+// Session represents the current logged in entities identity and
+// authentication objects which could be a Machine or User
+type Session interface {
+	Type() string
+	ID() *identity.ID
+	AuthID() *identity.ID
+	Username() string
+	Name() string
+	Email() string
+}
+
+func (s *session) Type() string {
+	return s.sessionType
+}
+
+func (s *session) ID() *identity.ID {
+	return s.identity.ID
+}
+
+func (s *session) AuthID() *identity.ID {
+	return s.auth.ID
+}
+
+func (s *session) Username() string {
+	if s.sessionType == apitypes.MachineSession {
+		return s.identity.Body.(*primitive.Machine).Name
+	}
+
+	return s.identity.Body.(*primitive.User).Username
+}
+
+func (s *session) Name() string {
+	if s.sessionType == apitypes.MachineSession {
+		return s.identity.Body.(*primitive.Machine).Name
+	}
+
+	return s.identity.Body.(*primitive.User).Name
+}
+
+func (s *session) Email() string {
+	if s.sessionType == apitypes.MachineSession {
+		return "none"
+	}
+
+	return s.identity.Body.(*primitive.User).Email
+}
+
+// NewSession returns a new session constructed from the payload of the current
+// identity as returned from the Daemon
+func NewSession(resp *apitypes.Self) (Session, error) {
```

Comment:
Since `NewSession` is already switching on the type here, how about making `Session` a struct instead of an interface, and pulling the appropriate values to put into the struct out here?